docs/lesson-plans/week-1.md

```diff
@@ -0,0 +1,649 @@
+is installed.
+- For Git bash, the terminal needs to be manually set in every project. Don't forget to do that, or
+else you'll end up running commands that don't exist in Windows.
+
+## Terminal Basics
+
+The terminal is a really important tool, which you'll find extremely useful throughout your 
+developer career. Not only does it allow you to access settings and commands which you would otherwise
+not have access to, it sometimes is just easier to be able to do some things by typing out commands
+rather than pointing and clicking with a mouse. 
+
+Learning how to use the terminal well may take a while, but we'll get you started with a few commands
+we think every developer should know. We'll also show you how to build your projects if you 
+haven't tried doing that yet.
+
+>Note: we'll be showing you how to use unix-based bash commands, which means that these commands will 
+>work for Mac and Linux computers. If you have a Windows computer, either download the Linux subsystem
+>or follow the commands in the backend README.md to install Git bash and start using that in your 
+>IntelliJ projects.
+
+### Common Commands
+
+#### cd
+
+`cd` is arguable one of the most used commands in the terminal. It stands for 'change directory', and it
+allows you to navigate around your computer. 
+
+Format:
+```shell script
+cd <directory>
+```
+
+- You can chain multiple directories together by adding '/' between each directory if you would like
+to jump to a specific one.
+- You can jump up directories with the '..' notation.
+- You can jump to your home directory (the one with Desktop, Documents, ... inside of it) by just typing `cd` or
+`cd ~/`
+
+Examples:
+```shell script
+# Go to your desktop
+cd Desktop
+
+# Go to back to your root directory
+cd ..
+
+# Go up multiple directories
+cd ../../../
+
+# Go back to your root directory
+cd ~/
+```
+
+#### ls
+
+`ls` is also a very commonly used commands. It lists all of the contents of a directory (both files and 
+sub-directories).
+
+Format:
+```shell script
+ls <optional filepath>
+```
+
+- You can list hidden files in a directory with the `-a` flag
+
+Examples:
+```shell script
+# List files in the current directory
+ls
+
+# List all files in the current directory
+ls -a
+
+# List files in the Desktop directory
+ls Desktop
+```
+
+#### pwd
+
+`pwd` stands for 'print working directory', and it logs the current directory you're located at.
+
+Format:
+```shell script
+pwd
+```
+
+#### mv
+
+`mv` is a command that lets you move a file or directory to another location. It can also be used
+to rename a file or directory (since that's the same thing as moving it in a way).
+
+Format:
+```shell script
+mv <from filepath> <to filepath>
+```
+
+- You can't move a file inside of itself or one of its subdirectories (without messing 
+around with symbolic links, but that's way outside of the material of this course).
```

Comment:
Add also applies to moving directories

Reply:
Yeah, that's included. Or do you want me to add more of an explanation.